### DISCUSSION TITLE
Warning about too little filament?

### DISCUSSION THREAD

**Discussion by Horgul123** [Donkie/Spoolman#271] [General]
Is there a solution available which could generate a warning (e.g. in mainsail) if the amount of filament on the active spool is lower than the calculated amount of filament needed?

Background: 
Yesterday I've forgotten to check how much filament is on the spool and startet a print. Unfortunately the manufacturer of the filament (Sunlu( decided to bend the end of the filament so that the printer was not able to throw if from the spool and I didn't get a "out of filament" warning. The printer "printed" the last one or two hours without filament....... :(
(I know that there are senors available which can recognize this issue, but I didn't find a was to install it on a Creality K1, there is no port available for attaching the "no filament movement" sensor)

**Comment by rufinus** [comment]
mainsail already does this. also if the material is not the same as in the filename.
![image](https://github.com/Donkie/Spoolman/assets/343160/5f0f6b9e-71bc-4040-8ca2-72fc9598fdbf)
![image](https://github.com/Donkie/Spoolman/assets/343160/eaf918b7-8682-4e68-b4db-bf9972e074fa)

**Reply by Horgul123** [reply]
Hmmm.. What happens if I don't click print? Will it start the print after a while?
As mentioned in my post: I run out of filament :(

Do I need to configure something? In Mainsail I can see the Spoolman widget with the active spool and the remaining filament.

**Reply by rufinus** [reply]
> I run out of filament :(

nobody said its an exact science

**Reply by Horgul123** [reply]
sure.... As I did not check how much filament was left and how much the slicer calculated I don't know if it should have warned me...

But thanks for the information, I will have a closer look the next time :)
